src/meds_torch/tune.py

```diff
@@ -81,31 +102,45 @@ def get_checkpoint_path(log_dir, checkpoint_dir_name, time_output_path):
         return checkpoint_files[0]
 
 
+def train_func(cfg):
+    if cfg.hparams_search.train_fn == "train":
+        initialize_objects = initialize_train_objects
+    elif cfg.hparams_search.train_fn == "finetune":
+        initialize_objects = initialize_finetune_objects
+    else:
+        raise ValueError(f"Invalid train_fn: {cfg.hparams_search.train_fn}, should be 'train' or 'finetune'")
+    plugins = [RayLightningEnvironment()]
+
+    object_dict = initialize_objects(cfg, plugins=plugins)
+    dm = object_dict["datamodule"]
+    model = object_dict["model"]
+    trainer = object_dict["trainer"]
+
+    trainer = prepare_trainer(trainer)
+    trainer.fit(model, datamodule=dm)
+
+
 @hydra.main(version_base="1.3", config_path=str(config_yaml.parent.resolve()), config_name=config_yaml.stem)
 def main(cfg: DictConfig) -> float | None:
     """Main entry point for training.
 
-    :param cfg: DictConfig configuration composed by Hydra.
-    :return: Optional[float] with optimized metric value.
+    Args:
+        cfg: DictConfig configuration composed by Hydra.
+    Returns:
+        Optional[float] with optimized metric value.
     """
+    os.environ["RAY_memory_monitor_refresh_ms"] = "0"
     # apply extra utilities
     os.makedirs(cfg.paths.time_output_dir, exist_ok=True)
```

Comment:
I would recommend setting `cfg.paths.time_output_dir` to a `Path` once with a variable, then using the `pathlib` approach to make the directory.

Reply:
I was considering in hydra switching:
```yaml
time_output_dir: ${paths.output_dir}/${now:%Y-%m-%d_%H-%M-%S_%f}
```

to 
```yaml
time_output_dir:
    _target_: pathlib.Path
    _args_:
        - ${paths.output_dir}/${now:%Y-%m-%d_%H-%M-%S_%f}
```

Thoughts @mmcdermott ?

Reply:
It will make your config file code harder to read but it isn't crazy. I think either is fine.

Reply:
This would complicate overriding as well, so it would be better to just use pathlib as needed in the codebase

Reply:
I added the `time_output_dir` conversion from str -> Path in the task_wrapper decorator, which applies it to all scripts